### PR TITLE
chore(node) increment node ver to 20

### PR DESCRIPTION
**Pull request by horymury**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -184,8 +184,8 @@
     }
   },
   "engines": {
-    "node": ">=14.0.0",
-    "npm": ">=7.0.0"
+    "node": ">=18.0.0",
+    "npm": ">=8.0.0"
```
Comment (saghul) [inline package.json:188]: Shouldn't this be 9?

**Review by saghul — COMMENTED** [review]
Left a comment, and there are still lint errors and warnings.

File: package.json (hunk 1 above)
Comment (horymury) [reply package.json:188]: I guess yes, for precisely 18.0 seems it still comes with 8.x, newer come with 9.x

Comment (saghul) [reply package.json:188]: Oh. I don't think 18.0 was LTS so I'd say requiring npm 9 is good.

**Comment by saghul** [comment]
Please also remember to update .nvmrc

**Comment by horymury** [comment]
> Left a comment, and there are still lint errors and warnings.

@saghul got rid of all warnings and errors

File: react/features/base/tooltip/actionTypes.tsx [hunk 2]
```diff
@@ -1,6 +1,5 @@
 /**
- * The type of the action which signals a tooltip is being displayed.
- *
+ * The type of the action which signals a tooltip is being displayed
```
Comment (saghul) [inline react/features/base/tooltip/actionTypes.tsx:2]: Does this give an error if you end it with a period?

File: react/features/connection-indicator/statsEmitter.ts [hunk 3]
```diff
@@ -6,8 +6,7 @@ import {
 } from '../base/lib-jitsi-meet';
 
 /**
- * Contains all the callbacks to be notified when stats are updated.
- *
+ * Contains all the callbacks to be notified when stats are updated
```
Comment (saghul) [inline react/features/connection-indicator/statsEmitter.ts:9]: Ditto

File: react/features/screenshot-capture/constants.ts [hunk 4]
```diff
@@ -22,17 +22,15 @@ export const SET_INTERVAL = 1;
 
 /**
  * CLEAR_INTERVAL constant is used to clear the interval and it is set in
- * the id property of the request.data property.
- *
+ * the id property of the request.data property
```
Comment (saghul) [inline react/features/screenshot-capture/constants.ts:25]: ditto.

File: react/features/screenshot-capture/constants.ts [hunk 5]
```diff
@@ -22,17 +22,15 @@ export const SET_INTERVAL = 1;
...
  * {
  *      id: CLEAR_INTERVAL
  * }.
  */
 export const CLEAR_INTERVAL = 2;
 
 /**
- * INTERVAL_TIMEOUT constant is used as response and it is set in the id property.
- *
+ * INTERVAL_TIMEOUT constant is used as response and it is set in the id property
```
Comment (saghul) [inline react/features/screenshot-capture/constants.ts:33]: ditto.

File: react/features/stream-effects/virtual-background/TimerWorker.ts [hunk 6]
```diff
@@ -13,17 +13,15 @@ export const SET_TIMEOUT = 1;
 
 /**
  * CLEAR_TIMEOUT constant is used to clear the interval and it is set in
- * the id property of the request.data property.
- *
+ * the id property of the request.data property
```
Comment (saghul) [inline react/features/stream-effects/virtual-background/TimerWorker.ts:16]: ditto.

File: react/features/stream-effects/virtual-background/TimerWorker.ts [hunk 7]
```diff
@@ -13,17 +13,15 @@ export const SET_TIMEOUT = 1;
...
  * {
  *      id: CLEAR_TIMEOUT
  * }.
  */
 export const CLEAR_TIMEOUT = 2;
 
 /**
- * TIMEOUT_TICK constant is used as response and it is set in the id property.
- *
+ * TIMEOUT_TICK constant is used as response and it is set in the id property
```
Comment (saghul) [inline react/features/stream-effects/virtual-background/TimerWorker.ts:24]: ditto

**Review by saghul — APPROVED** [review]
LGTM with a few questions. Please don't merge this yet, we need to update Jenkins first.

**Comment by horymury** [comment]
> LGTM with a few questions. Please don't merge this yet, we need to update Jenkins first.

Got it. About the questions, with . it shows warning
image

**Comment by saghul** [comment]
I mean in the complete sentences. If ending them in a period creates a warning then something is wrong, because we used to enforce that.

**Comment by horymury** [comment]
> I mean in the complete sentences. If ending them in a period creates a warning then something is wrong, because we used to enforce that.

I think it was seeing the . after the closing bracket as the end of sentence.

**Comment by saghul** [comment]
Ok, so can you please add it back to the full sentences then?

**Comment by horymury** [comment]
> Ok, so can you please add it back to the full sentences then?

@saghul I tried having the . after the sentence/after the closing bracket/both places, nothing worked.
What worked and I think looks ok is marking the object as ```code```:
image

File: react/features/base/config/functions.web.ts [hunk 8]
```diff
@@ -58,7 +58,7 @@ export function getWebHIDFeatureConfig(state: IReduxState): boolean {
 /**
  * Checks if the specified button is enabled.
  *
- * @param {string} buttonName - The name of the button.
+ * @param {string} buttonName - The name of the button
```
Comment (saghul) [inline react/features/base/config/functions.web.ts:61]: I think this should be: `@param {string} buttonName - The name of the button. See: {@link interfaceConfig}.` adn the linter should be happy.

**Comment by robertpin** [comment]
The issue with . is from the new version of `eslint-plugin-jsdoc`. I noticed that it behaves differently back when I updated all the other eslint plugins so I left this one on the old version.
FYI

**Comment by saghul** [comment]
> The issue with . is from the new version of `eslint-plugin-jsdoc`. I noticed that it behaves differently back when I updated all the other eslint plugins so I left this one on the old version. FYI

Thanks for taking a look! Yep I noticed that too. Alas we need to update in order to avoid warnings regarding the supported Node version.

**Comment by horymury** [comment]
> 

@robertpin  thank you, yep I supected the same - but as @saghul  mentioned we need to update. 
Is the ```code``` workaround a good enough solution? 
Thank you

**Comment by saghul** [comment]
> > 
> 
> @robertpin thank you, yep I supected the same - but as @saghul mentioned we need to update. Is the `code` workaround a good enough solution? Thank you

The code solution WFM.

**Review by saghul — APPROVED** [review]
LGTM but pl wait on merging as I mentioned earlier.

**Comment by saghul** [comment]
Sigh, gotta get back to this...

**Comment by saghul** [comment]
@horymury can you update this to 20 since that made it LTS?

File: modules/UI/videolayout/LargeVideoManager.js [hunk 9]
```diff
@@ -3,7 +3,7 @@
 import Logger from '@jitsi/logger';
 import $ from 'jquery';
 import React from 'react';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
```
Comment (saghul) [inline modules/UI/videolayout/LargeVideoManager.js:6]: Why are we doing this? I think this opts us into concurrent roots, which I don't think we are ready to do just yet. Let's keep this just about Node please.

**Review by saghul — CHANGES_REQUESTED** [review]

**Comment by horymury** [comment]
> We need to switch to 20 now...

@saghul Updated the PR for node 20. Think you can help with the jenkins machines? Thank you.

File: modules/UI/videolayout/LargeVideoManager.js (hunk 9 above)
Comment (horymury) [reply modules/UI/videolayout/LargeVideoManager.js]: This was in order to fix :
image
I reverted for now though to keep this PR for node only.

**Review by saghul — APPROVED** [review]
LGTM! Pl don't merge this yet,

**Comment by damencho** [comment]
Jenkins, test this please.

**Pull request merged**